frontend/model/contracts/group.js

```diff
@@ -312,32 +312,138 @@ sbp('chelonia/defineContract', {
     groupMincomeCurrency (state, getters) {
       return getters.groupSettings.mincomeCurrency
     },
+    // Oldest period key first.
+    groupSortedPeriodKeys (state, getters) {
+      // The .sort() call might be only necessary in older browser which don't maintain object key ordering.
+      // A comparator function isn't required for now since our keys are ISO strings.
+      return Object.keys(getters.currentGroupState.paymentsByPeriod ?? {}).sort()
+    },
+    // Returns either the known period stamp for the given date,
+    // or the predicted one according to the period length.
+    // May return 'undefined', in which case the caller should check archived data.
     periodStampGivenDate (state, getters) {
       return (recentDate: string | Date) => {
-        if (typeof recentDate !== 'string') {
-          recentDate = recentDate.toISOString()
-        }
+        if (typeof recentDate !== 'string') recentDate = recentDate.toISOString()
+        if (!isPeriodStamp(recentDate)) throw new TypeError('must be date or isostring')
         const { distributionDate, distributionPeriodLength } = getters.groupSettings
-
-        if (!distributionDate) return null
-
-        return periodStampGivenDate({
-          recentDate,
-          periodStart: distributionDate,
-          periodLength: distributionPeriodLength
-        })
+        if (!distributionDate) return
```

Comment:
Havig APIs wich return `null` rather than `undefined` is generally considered a bad idea AFAIK, and the change was easy to do

Reply:
> Having APIs wich return null rather than undefined is generally considered a bad idea AFAIK

🤔 Really? Why is that? I'm used to C and Java where returning `null` or `NULL` is common. It's JS that's weird with its two different "null" values...

Reply:
IMO that has mainly to do with the handling of default function parameters.
For example if you have a `foo (arg = defaultValue) => { ... } function`, then `defaultValue` will be used if `arg` is undefined, but won't if `arg` is null. Then inside the function you'll likely have to check again whether `arg` is null.

This [link](https://www.split.io/blog/bias-against-null-javascript/) is also interesting

Reply:
That's a great link! Thanks for sharing!